File: tests/nonsmoke/functional/CompileTests/OpenMP_tests/parallel-if.c
// Test if clause handling
// number of threads should be set to 1 if the if-clause's expression evaluates to be false
#include <assert.h>
#include <stdio.h>
#include <omp.h>

int main(void)
{
  int i=0;
#pragma omp parallel if(i == 0)
  {
    printf("Mutual exclusive output 1.\n");
  }

#pragma omp parallel if(i != 0)
  {
#pragma omp single
    {
      assert (omp_get_num_threads() == 1 );
    }
    printf("Mutual exclusive output 2.\n");
  }
 
  return 0;
}

File: tests/nonsmoke/functional/CompileTests/OpenMP_tests/fortran/parallel-if.f
C***************************************************
C test if clause
C Liao, 3/17/2011
C***************************************************
      PROGRAM HELLO
      include "omp_lib.h"
      integer i
      integer tcount
      i = 0
!$OMP PARALLEL if( i .eq. 0)
      PRINT *, 'Hello World!'
c$omp end parallel 

!$OMP PARALLEL if( i .ne. 0)
      PRINT *, 'Hello World 2!'
!$omp single
      tcount = omp_get_num_threads ()
      print *, tcount
      if (tcount .ne. 1 ) call abort
!$omp end single

c$omp end parallel 
      PRINT *, 'the end'
      END
